### PR TITLE
chore: Migrates RefreshChartOverlay into Chart

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Migrates `RefreshChartOverlay` into `Chart`.

`RefreshChartOverlay` wasn't used anywhere outside `Chart` and it's a specific behavior of the chart component. Also, most of its properties weren’t being used.

### TEST PLAN
1 - Execute `Chart` tests
2 - All tests should pass

@rusackas 

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/chart/Chart.jsx [hunk 1]
```diff
@@ -175,7 +186,6 @@ class Chart extends React.PureComponent {
 
   render() {
     const {
-      width,
```
Comment (geido) [inline superset-frontend/src/chart/Chart.jsx:178]: It looks like the width was used in a few places and it is also defined in the PropTypes. Would you like to check where it is used and delete it if you deem it unnecessary?

**Review by geido — CHANGES_REQUESTED** [review]

File: superset-frontend/src/chart/Chart.jsx (hunk 1 above)
Comment (michael-s-molina) [reply superset-frontend/src/chart/Chart.jsx:178]: Yes. I'll do that.

**Comment by junlincc** [comment]
added `test:component` label to track project, but not actually related. Thanks for the PR and review @michael-s-molina @geido 🙏

**Comment by michael-s-molina** [comment]
@rusackas It seems that the Cypress error in `explore/control.test.ts` is not related to this PR or maybe it's not even an error but the random nature of some tests 😆.

It executes successfully locally:

Screen Shot 2021-02-23 at 3 44 20 PM

I probably spammed reviewers' emails with forced pushes or open/close trying to make this pass (sorry about that). Maybe you are luckier than me? Or maybe it's an error and I'm low on coffee 😩?

File: superset-frontend/src/chart/Chart.jsx [hunk 2]
```diff
@@ -19,12 +19,12 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import Alert from 'src/components/Alert';
-import { styled, logging } from '@superset-ui/core';
+import { styled, logging, t } from '@superset-ui/core';
 
 import { isFeatureEnabled, FeatureFlag } from 'src/featureFlags';
-import { Logger, LOG_ACTIONS_RENDER_CHART } from '../logger/LogUtils';
+import Button from 'src/components/Button';
 import Loading from '../components/Loading';
```
Comment (rusackas) [inline superset-frontend/src/chart/Chart.jsx:26]:
```suggestion
import Loading from 'src/components/Loading';
```
might as well add `src` alias paths for the rest of these imports while we're here (I won't point out the rest... you get the idea) :)

**Comment by michael-s-molina** [comment]
I finally found out what was causing the test to fail. Although `width` was no longer used in `Chart`, it was used in `ChartRenderer` and therefore could not be removed from `ChartContainer`. Turns out it was the lack of coffee 😬

**Review by rusackas — APPROVED** [review]
LGTM! Just needs a rebase to merge.

**Review by rusackas — APPROVED** [review]

**Pull request merged**